homeassistant/components/sensor/__init__.py

```diff
@@ -422,9 +421,9 @@ def state(self) -> Any:
                     f"non-numeric device class: {device_class}"
                 )
 
-        # Checks below only apply if there is a value
-        if value is None:
-            return None
+        # Checks below only apply if there is a value and a device_class
+        if value is None or device_class is None:
```

Comment:
The numeric checks can also be done when there is no unit, no device class, but a state class is defined. State classes means it is numeric too.

Reply:
(PS: Sorry this was meant to be a generic comment on the PR :S)

Reply:
This check has been adjusted and moved down.